src/rpc_extra.rs

```diff
@@ -18,22 +18,12 @@ where
     let epoch_schedule = client.get_epoch_schedule()?;
     let first_slot = epoch_schedule.get_first_slot_in_epoch(epoch);
 
-    // We cannot use an excessively large range if the epoch just started. There is a chance that
-    // the end slot has not been reached and strange behaviour will occur.
-    // If this is the current epoch and less than `SLOT_OFFSET` slots have elapsed, then do not define an
-    // end_slot for use in the RPC call.
-    let end_slot = if epoch_info.epoch == epoch && epoch_info.slot_index < SLOT_OFFSET {
-        None
-    } else {
-        Some(first_slot + SLOT_OFFSET)
-    };
-
     // First block in `epoch`.
-    let first_block = client.get_blocks(first_slot, end_slot)?.get(0).cloned();
+    let first_block = client.get_blocks_with_limit(first_slot, 1)?.get(0).cloned();
 
     if let Some(block) = first_block {
         f(block)
-    } else if end_slot.is_none() {
+    } else if epoch_info.slot_index < SLOT_OFFSET {
```

Comment:
From what I understand, if the slot index is less than the offset then we don't return an error, in the case that there are no blocks?

Reply:
Yes, this is to simulate the previous behaviour where we returned an error only when there were no blocks after `SLOT_OFFSET` was reached.

Reply:
As far as I can tell the slot_offset error is always bubbled up the user, so we won't break anything if we change this. Is there a better API design, perhaps one that doesn't use the `SLOT_OFFSET` at all?

Reply:
We can never return a "no blocks found" error for example.

Reply:
Yes, I think the approach in a14883d is quite nice. Let the error be reserved for more serious errors, and treat the lack of a first block as a `None`.